pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/AuxClassPathController.java

```diff
@@ -0,0 +1,204 @@
+    private Button setClassPathButton;
+    @FXML
+    private Button cancelButton;
+
+
+    public AuxClassPathController(ObservableList<File> auxClassPathFiles, DesignerRoot designerRoot) {
+        this.designerRoot = designerRoot;
+
+        if (auxClassPathFiles != null) {
+            fileListView.setItems(auxClassPathFiles);
+        }
+
+        try {
+            showAuxPathWizard();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+
+        selectFilesButton.setOnAction(e -> onSelectFileClicked());
+        removeFileButton.setOnAction(e -> onRemoveFileClicked());
+        setClassPathButton.setOnAction(e -> {
+            try {
+                setClassPath(classPathGenerator());
+            } catch (IOException e1) {
+                e1.printStackTrace();
+            }
+        });
+
+
+        moveItemUpButton.setOnAction(e -> moveUp());
+        moveItemDownButton.setOnAction(e -> moveDown());
+        cancelButton.setOnAction(e -> closePopup());
+    }
+
+
+    private void onSelectFileClicked() {
+        FileChooser chooser = new FileChooser();
+        chooser.setTitle("Select Files");
+        chooser.getExtensionFilters().addAll(
+            new FileChooser.ExtensionFilter("Java JARs", "*.jar"),
+            new FileChooser.ExtensionFilter("Java WARs", "*.war"),
+            new FileChooser.ExtensionFilter("Java EARs", "*.ear"),
+            new FileChooser.ExtensionFilter("Java class files", "*.class")
+        );
+        List<File> files = chooser.showOpenMultipleDialog(designerRoot.getMainStage());
+        fileListView.getItems().addAll(files);
+    }
+
+
+    private void onRemoveFileClicked() {
+        File f = fileListView.getSelectionModel().getSelectedItem();
+        fileListView.getItems().remove(f);
+    }
+
+
+    private void moveUp() {
+        moveItem(-1);
+    }
+
+
+    private void moveDown() {
+        moveItem(1);
+    }
+
+
+    public void moveItem(int direction) {
+        // Checking selected item
+        if (fileListView.getSelectionModel().getSelectedItem() == null) {
+            return;
+        }
+
+        // Calculate new index using move direction
+        int newIndex = fileListView.getSelectionModel().getSelectedIndex() + direction;
+
+        if (newIndex < 0 || newIndex >= fileListView.getItems().size()) {
+            return;
+        }
+
+        File selected = fileListView.getSelectionModel().getSelectedItem();
+
+        // Removing removable element
+        fileListView.getItems().remove(selected);
+        // Insert it in new position
+        fileListView.getItems().add(newIndex, selected);
+        //Restore Selection
+        fileListView.scrollTo(newIndex);
+        fileListView.getSelectionModel().select(newIndex);
+
+    }
+
+
+    private void setValidationSupport() {
+
+    }
+
+    private String classPathGenerator() throws IOException {
```

Comment:
This shouldn't be here. You should use a similar routine to map the auxclasspathFiles to the auxclasspathClassLoader in SourceEditorController